src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCheck.java

```diff
@@ -132,14 +132,12 @@ public void visitToken(DetailAST ast) {
      */
     private static boolean isInEmptyForInitializerOrCondition(DetailAST semicolonAst) {
         boolean result = false;
-        if (semicolonAst.getType() == TokenTypes.SEMI) {
```

Comment:
Conditions inside this if ensures this is the semi without checking for it.
Regression couldn't find me anything to work with.